sklearn/metrics/cluster/unsupervised.py

```diff
@@ -205,6 +209,8 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
 
     sil_samples = inter_clust_dists - intra_clust_dists
     sil_samples /= np.maximum(intra_clust_dists, inter_clust_dists)
```

Comment:
If all samples belong to one cluster, `silhouette_score` is undefined according to the paper. All points being identical is not covered in the paper. Understandably.